Sources/Basics/Concurrency/ConcurrencyHelpers.swift

```diff
@@ -76,3 +76,33 @@ extension DispatchQueue {
         }
     }
 }
+
+package actor ConcurrencyLimiter {
```

Comment:
Can you instead copy https://github.com/swiftlang/swift-build/blob/main/Sources/SWBUtil/AsyncOperationQueue.swift? We should avoid having slightly different implementations for the same problem; also that implementation is safer because it guards against unbalanced acquire/release by only providing a closure-based API.

Reply:
Done.